src/transformers/integrations/hqq.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2024 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"HQQ (Half-Quadratic Quantization) integration file"
+
+from ..utils import is_hqq_available, is_torch_available, logging
+
+
+if is_torch_available():
+    import torch
+else:
+    torch = None
+
+if is_hqq_available():
+    from hqq.core.quantize import HQQLinear
+else:
+    HQQLinear = None
```

Comment:
```suggestion
```
This can be removed now since hqq is imported locally in `get_linear_tags`

Reply:
https://github.com/huggingface/transformers/pull/29637/commits/b54e87b204fb72941ac081c54ad32611a3918543